src/wallet/scriptpubkeyman.h

```diff
@@ -286,7 +286,9 @@ class LegacyScriptPubKeyMan : public ScriptPubKeyMan, public FillableSigningProv
     WatchOnlySet setWatchOnly GUARDED_BY(cs_KeyStore);
     WatchKeyMap mapWatchKeys GUARDED_BY(cs_KeyStore);
 
-    int64_t nTimeFirstKey GUARDED_BY(cs_KeyStore) = 0;
+    // By default, do not scan any block until keys/scripts are generated/imported
+    int64_t NO_KEY_TIME = std::numeric_limits<int64_t>::max();
```

Comment:
Seems like `NO_KEY_TIME` ought to be `0` so it triggers a full scan

Reply:
> Seems like `NO_KEY_TIME` ought to be `0` so it triggers a full scan

That was the bug I solved 0a1107d0.
By default, the legacy spkm is empty. It has no keys nor scripts, thus it should not trigger a full scan.
Once the first key is generated or the first script is imported,
the legacy spkm will update the `nTimeFirstKey` time automatically and start scanning blocks.